### PR TITLE
[#2017] feat(coordinator/server): Support config to exclude verbose operation from rpc audit log

### LINKED ISSUE #2017: [Improvement] Support config to exclude verbose operation from rpc audit log
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

upport configs to specify a list of operation name to be an exclude list, when audit rpc call to audit log, check whether the operations contained in this exclude list, if so, ignore it.

### How should we improve?

For example, appheartbeat and heartbeat from node are verbose and waste disk space. As we don't know who want to specify another operation to exclude, so I defined this as a list type config.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Support  configs to specify a list of operation name to be an exclude list, when audit rpc call to audit log, check whether the operations contained in this exclude list, if so, ignore it.

### Why are the changes needed?

For example, appheartbeat and heartbeat from node are verbose and waste disk space. As we don't know who want to specify another operation to exclude, so I defined this as a list type config.

Fix #2017

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

Start coordinator and shuffle server and start a spark app, check the rpc auditlog.

- coordinator audit log
```
[2024-08-06 11:39:00.921] cmd=fetchClientConfV2	statusCode=SUCCESS	appId=N/A	shuffleId=-1	executionTimeUs=2763	
[2024-08-06 11:39:27.689] cmd=getShuffleAssignments	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=11574	args{partitionNum=16, partitionNumPerRange=1, replica=1, requiredTags=[GRPC_NETTY, ss_v5], requiredShuffleServerNumber=-1, faultyServerIds=[], stageId=-1, stageAttemptNumber=0, isReassign=false}
[2024-08-06 11:39:28.238] cmd=registerApplicationInfo	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=-1	executionTimeUs=1826	args{user=user}
```

- server audit log
```
[2024-08-06 11:39:28.200] cmd=registerShuffle	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=46804	args{remoteStoragePath=, user=root, stageAttemptNumber=0}
[2024-08-06 11:39:30.413] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=33176	
[2024-08-06 11:39:30.411] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=32673	
[2024-08-06 11:39:30.414] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=31966	
[2024-08-06 11:39:30.413] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=31169	
[2024-08-06 11:39:30.411] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=27557	
[2024-08-06 11:39:30.412] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=27368	
[2024-08-06 11:39:30.413] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=27773	
[2024-08-06 11:39:30.414] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=36704	
[2024-08-06 11:39:30.415] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=36157	
[2024-08-06 11:39:30.415] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=37076	
[2024-08-06 11:39:30.415] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=37601	
[2024-08-06 11:39:30.416] cmd=requireBuffer	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=31976	args{requireSize=233, partitionIdsSize=1}
[2024-08-06 11:39:30.416] cmd=requireBuffer	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=37132	args{requireSize=233, partitionIdsSize=1}
[2024-08-06 11:39:30.416] cmd=requireBuffer	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=37949	args{requireSize=233, partitionIdsSize=1}
[2024-08-06 11:39:30.416] cmd=requireBuffer	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=37973	args{requireSize=233, partitionIdsSize=1}
[2024-08-06 11:39:30.416] cmd=requireBuffer	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=37058	args{requireSize=233, partitionIdsSize=1}
[2024-08-06 11:39:30.598] cmd=sendShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=88847	args{requireBufferId=5, requireSize=233, isPreAllocated=true, requireBlocksSize=39, stageAttemptNumber=0}
[2024-08-06 11:39:30.598] cmd=sendShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=89060	args{requireBufferId=2, requireSize=233, isPreAllocated=true, requireBlocksSize=39, stageAttemptNumber=0}
[2024-08-06 11:39:30.603] cmd=sendShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=709	args{requireBufferId=3, requireSize=233, isPreAllocated=true, requireBlocksSize=39, stageAttemptNumber=0}
[2024-08-06 11:39:30.604] cmd=sendShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=273	args{requireBufferId=1, requireSize=233, isPreAllocated=true, requireBlocksSize=39, stageAttemptNumber=0}
[2024-08-06 11:39:30.605] cmd=sendShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=226	args{requireBufferId=4, requireSize=233, isPreAllocated=true, requireBlocksSize=39, stageAttemptNumber=0}
[2024-08-06 11:39:30.765] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=1076	
[2024-08-06 11:39:30.765] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=5471	
[2024-08-06 11:39:30.765] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=5029	
[2024-08-06 11:39:30.766] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=5571	
[2024-08-06 11:39:30.770] cmd=commitShuffleTask	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=8312	
[2024-08-06 11:39:31.011] cmd=reportShuffleResult	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=30035	args{taskAttemptId=60, bitmapNum=1, partitionToBlockIdsSize=1}
[2024-08-06 11:39:31.011] cmd=reportShuffleResult	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=30951	args{taskAttemptId=48, bitmapNum=1, partitionToBlockIdsSize=1}
[2024-08-06 11:39:31.011] cmd=reportShuffleResult	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=30146	args{taskAttemptId=24, bitmapNum=1, partitionToBlockIdsSize=1}
[2024-08-06 11:39:31.011] cmd=reportShuffleResult	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=30311	args{taskAttemptId=12, bitmapNum=1, partitionToBlockIdsSize=1}
[2024-08-06 11:39:31.875] cmd=finishShuffle	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=1039766	
[2024-08-06 11:39:32.211] cmd=reportShuffleResult	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=1190	args{taskAttemptId=36, bitmapNum=1, partitionToBlockIdsSize=1}
[2024-08-06 11:39:32.942] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=40910	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.944] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=40852	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.944] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=40974	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=46557	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=46859	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=46765	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=47568	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=24175	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=45910	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=47714	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.947] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=38633	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.948] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=46078	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.948] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=47629	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.948] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=47867	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.948] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=47450	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:32.953] cmd=getShuffleResultForMultiPart	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=48200	args{partitionsListSize=1, blockIdLayout=blockIdLayout[seq: 21 bits, part: 20 bits, task: 22 bits]}
[2024-08-06 11:39:33.110] cmd=getLocalShuffleIndex	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=43637	args{partitionId=1, partitionNumPerRange=1, partitionNum=16}
[2024-08-06 11:39:33.114] cmd=getLocalShuffleIndex	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=1599	args{partitionId=3, partitionNumPerRange=1, partitionNum=16}
[2024-08-06 11:39:33.115] cmd=getLocalShuffleIndex	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=725	args{partitionId=2, partitionNumPerRange=1, partitionNum=16}
[2024-08-06 11:39:33.130] cmd=getLocalShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=2649	args{partitionId=2, partitionNumPerRange=1, partitionNum=16, offset=0, length=14}
[2024-08-06 11:39:33.130] cmd=getLocalShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=2551	args{partitionId=3, partitionNumPerRange=1, partitionNum=16, offset=0, length=7}
[2024-08-06 11:39:33.131] cmd=getLocalShuffleData	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=0	executionTimeUs=522	args{partitionId=1, partitionNumPerRange=1, partitionNum=16, offset=0, length=14}
[2024-08-06 11:39:48.148] cmd=unregisterShuffleByAppId	statusCode=SUCCESS	appId=app-20240806113903-0010_1722915541508	shuffleId=-1	executionTimeUs=14607	
```

### CODE REVIEW

**Comment by jerqi** [comment]
You add a config option. This is a user-facing change. You should add the document. And you should also create an issue to track it.

**Comment by maobaolong** [comment]
@jerqi Thanks for your change suggestion! I've create an issue track this, and I've add the config to `docs/coordinator_guide.md` and `docs/server_guide.md`

File: coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorConf.java [hunk 1]
```diff
@@ -250,6 +250,12 @@ public class CoordinatorConf extends RssBaseConf {
           .defaultValue(true)
           .withDescription(
               "When set to true, for auditing purposes, the coordinator will log audit records for every rpc request operation. ");
+  public static final ConfigOption<List<String>> COORDINATOR_RPC_AUDIT_LOG_EXCLUDE_LIST =
+      ConfigOptions.key("rss.coordinator.rpc.audit.log.exclude.list")
```
Comment (jerqi) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/CoordinatorConf.java:254]: `exclude.list`->`excludeList`

**Comment by maobaolong** [comment]
@jerqi Good suggestion. I've renamed and committed. PTAL.

**Review by jerqi — APPROVED** [review]

**Review by zuston — APPROVED** [review]

**Comment by maobaolong** [comment]
@jerqi @zuston Thanks for your review!

**Pull request merged**